### PR TITLE
[4.6.x] IPROTO-283 Transcode bytes properties from JSON array

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/IPROTO-283

Backport or #224

### CODE REVIEW

File: core/src/test/java/org/infinispan/protostream/ProtobufUtilTest.java [hunk 1]
```diff
@@ -224,6 +225,28 @@ public void testJsonWithDifferentFieldOrder() throws Exception {
...
+      Item java2 = ProtobufUtil.fromWrappedByteArray(context, proto2);
+      assertEquals(java, java2);
+
+      // Moreover, we want to parse also Java byte[] / Proto bytes fields in the JSON format `[7,7,7]`
+      String alternativeJson = "{\"_type\":\"Item\",\"code\":\"c7\",\"byteVector\":[7,7,7],\"floatVector\":[1.1,1.1,1.1],\"buggy\":\"bla bla bla\"}";
+      byte[] proto3 = ProtobufUtil.fromCanonicalJSON(context, new StringReader(alternativeJson));
+      Item java3 = ProtobufUtil.fromWrappedByteArray(context, proto3);
+      assertEquals(java, java3);
+   }
+
```
Comment (rigazilla) [inline core/src/test/java/org/infinispan/protostream/ProtobufUtilTest.java:249]: Is there a test around to verify that the mapping between json array of int -> java int[] still works?
I can't find it

Comment (fax4ever) [reply core/src/test/java/org/infinispan/protostream/ProtobufUtilTest.java:249]: good idea. I added a int[] field to cover the case

**Comment by fax4ever** [comment]
I just pushed the changes from the main branch

**Comment by ryanemerson** [comment]
Thanks @fax4ever

**Comment by fax4ever** [comment]
Thanks!

**Pull request merged**